gqcp/include/QCModel/HF/UHF.hpp

```diff
@@ -320,10 +345,22 @@ class UHF {
      */
     OneDM<Scalar> calculateScalarBasis1DM(const Spin sigma) const {
```

Comment:
This method should return a `SpinResolvedOneDM`.

Reply:
Same question as for the `calculateOrthonormalBasis1DM` API.

Reply:
Same answer, here. You can remove the 'old' functionality that was meant to calculate only the alpha or beta part, and replace it with the new functionality that calculates them both and encapsulates them correctly.